Address CRAN complaints arising from test for #69, a data.table induced problem

benthestatistician opened the pull request:
Sets up a file for tests that we'd like to run, but that we don't want to send to CRAN when we release.  

To stay ahead of regressions, we want to test liberally.  But those sorts of tests are among the most likely to cause trouble with CRAN submissions, because often they involve provoking behavior from another package that differs from what CRAN wants to see. Here is a proposed solution.

My hope is that Travis will still run the tests that have been parked in tests/testthat/test.notforCRAN.R -- I'd appreciate a sound-off from someone who knows a bit about how Travis works.   I'm soliciting Josh's review partly for this reason and partly b/c he's the author of the regression test that this strategy avoids commenting out. 

This is a counter-proposal to the response  to the following email (from Kurt H) that @jwbowers had proposed in #93. 

>These seem to have undeclared package dependencies in their unit test code (R files in tests subdirs), see below.
>
>Can you pls fix as necessary?  (Add the missing package dependencies to Suggests, I guess.)
>
> Please note that these issues are currently not yet detected by the CRAN incoming (or regular) checks.
>
>Best
> -k

```
$RItools
'library' or 'require' call not declared from: ‘data.table’
```

josherrickson commented:
As far as I know or can tell, the strategy should work. I think we should test this theory (add a failing test to the file and commit, see if Travis catches it), but we'll need to fix up the tests that are already failing first.

benthestatistician commented:
NB: This is a request to pull into the master branch.  The master branch has a bunch more stuff that's not ready for CRAN.  This material also needs to be merged into whatever branch we'll be using more immediately to submit a patched RItools from.

benthestatistician commented:
@jwbowers the branch we last pushed to CRAN, as version 0.1-15, is issue71.  I've just now updated that branch with the contents of the commit discussed in this merge request, also updated its version no to 0.1-16.  With some CRAN comments for Kurt H it should be ready to go. (As Josh notes, the master branch is not close to ready for posting to CRAN, there's much cleanup to be done there yet.)
